Core/2.Processing/Jobs/GenerateSurfaceTrianglesJob.cs

```diff
@@ -48,271 +43,301 @@ struct CompareSortByBasePlaneIndex : System.Collections.Generic.IComparer<Chisel
-            for (int l = 0; l < surfaceLoopCount; l++)
+            // Read loop infos and edges
+            var loopCount = input.Read<int>();
+            NativeCollectionHelpers.EnsureMinimumSizeAndClear(ref surfaceLoopAllInfos, loopCount);
+            NativeCollectionHelpers.EnsureSizeAndClear(ref surfaceLoopAllEdges, loopCount);
+            for (int l = 0; l < loopCount; l++)
             {
                 surfaceLoopAllInfos[l] = input.Read<SurfaceInfo>();
-                var edgeCount   = input.Read<int>();
+                var edgeCount = input.Read<int>();
                 if (edgeCount > 0)
-                { 
-                    var edgesInner  = new UnsafeList<Edge>(edgeCount, Allocator.Temp);
+                {
+                    var edges = new UnsafeList<Edge>(edgeCount, Allocator.Temp);
                     for (int e = 0; e < edgeCount; e++)
-                    {
-                        edgesInner.AddNoResize(input.Read<Edge>());
-                    }
-                    surfaceLoopAllEdges[l] = edgesInner;
+                        edges.AddNoResize(input.Read<Edge>());
+                    surfaceLoopAllEdges[l] = edges;
                 }
             }
             input.EndForEachIndex();
 
-
-
             if (!basePolygonCache[brushNodeOrder].IsCreated)
                 return;
 
-            var maxLoops = 0;
-            var maxIndices = 0;
+            // Compute maximum sizes
+            int maxLoops = 0, maxIndices = 0;
             for (int s = 0; s < surfaceLoopIndices.Length; s++)
             {
-                if (!surfaceLoopIndices[s].IsCreated)
-                    continue;
-                var length = surfaceLoopIndices[s].Length;
-                maxIndices += length;
-                maxLoops = math.max(maxLoops, length);
+                if (!surfaceLoopIndices[s].IsCreated) continue;
+                var len = surfaceLoopIndices[s].Length;
+                maxIndices += len;
+                maxLoops = math.max(maxLoops, len);
             }
 
-
             ref var baseSurfaces            = ref basePolygonCache[brushNodeOrder].Value.surfaces;
-            var brushTransformations        = transformationCache[brushNodeOrder];
-            var treeToNode                  = brushTransformations.treeToNode;
-            var nodeToTreeInverseTransposed = math.transpose(treeToNode);
-            
-			Vertex2DRemapper vertex2DRemapper = new()
-			{
-				lookup      = new NativeList<int>(64, Allocator.Temp),
-				positions2D = new NativeList<double2>(64, Allocator.Temp),
-				edgeIndices = new NativeList<int>(128, Allocator.Temp)
-			};
-			UniqueVertexMapper uniqueVertexMapper = new()
+            var transform                    = transformationCache[brushNodeOrder];
+            var treeToNode                   = transform.treeToNode;
+            var nodeToTreeInvTrans          = math.transpose(treeToNode);
```

Comment:
isn't nodeToTree already inverse? wouldn't this be treeToNodeInv, or simply nodeToTree?

Reply:
oh I made the mistake myself lol